src/shared/components/common/dropdown.tsx

```diff
@@ -0,0 +1,52 @@
+import { Component, linkEvent } from "inferno";
+
+interface DropdownProps {
+  button: any
+  children: any
+}
+
+interface DropdownState {
+  showDropdown : boolean
+}
+
+export class Dropdown extends Component<DropdownProps, DropdownState> {
+  private emptyState: DropdownState = {
+    showDropdown: false,
+  };
+
+  constructor(props: any, context: any) {
+    super(props, context);
+    this.state = this.emptyState;
+  }
+
+  render() {
+    return (
+      <>
+        <span className="dropdown">
+        <button className="btn btn-link dropdown-toggle" 
+                onClick={linkEvent(this, this.handleToggleDropdown)}
+                role="button"
+                type="button"
+                aria-expanded="false">{this.props.button}</button>
+        {this.state.showDropdown && (
+        <div
+          class="dropdown-content navbar-nav"
+          onMouseLeave={linkEvent(
+            this,
+            this.handleToggleDropdown
+          )}
+        >
+          {this.props.children}
+        </div>
+        )}
+        </span>
+      </>
+    );
+  }
+
+  handleToggleDropdown(i: Dropdown, event: any) {
+    event.preventDefault();
+    i.state.showDropdown = !i.state.showDropdown;
+    i.setState(i.state);
+  }
+}
```

Comment:
None of this is necessary, check out the search.tsx for how to do user searching and filtering. Don't bother with array filtering, an `All` and then the specific types are enough.